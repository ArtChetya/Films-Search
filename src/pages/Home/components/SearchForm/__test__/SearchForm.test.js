import { mount, shallow } from 'enzyme/build';
import React from 'react';
import { SearchForm } from '..';

describe('SearchForm component', () => {
    let search;
    let searchBy;
    let onSearch;

    beforeAll(() => {
        search = '';
        searchBy = 'title';
        onSearch = jest.fn();
    });

    it('should be rendered correctly', () => {
        const component = shallow(
            <SearchForm
                search={search}
                searchBy={searchBy}
                onSearch={onSearch}
            />
        );
        expect(component).toMatchSnapshot();
    });

    it('should change value of input on change', () => {
        const expectedValue = 'someValue';
        const component = mount(
            <SearchForm
                search={search}
                searchBy={searchBy}
                onSearch={onSearch}
            />
        );

        component.find('input').simulate('change', {
            target: { value: expectedValue }
        });

        expect(component.find('input').props().value).toEqual(expectedValue);
    });

    it('should call onSearch of form submit', () => {
        const component = mount(
            <SearchForm
                search={search}
                searchBy={searchBy}
                onSearch={onSearch}
            />
        );
        component.find('form').simulate('submit');

        expect(onSearch).toHaveBeenCalled();
    });
});
