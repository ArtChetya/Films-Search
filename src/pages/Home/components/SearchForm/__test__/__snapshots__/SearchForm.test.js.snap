// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should be rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="space-between"
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <TransformText
        transform="uppercase"
        variant="h6"
      >
        find your movie
      </TransformText>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <SearchForm__SearchField
        fullWidth={true}
        margin="dense"
        onChange={[Function]}
        value=""
        variant="filled"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={5}
    >
      <SearchBy
        onOptionChange={[Function]}
        options={
          Array [
            Object {
              "id": "title",
              "label": "Title",
            },
            Object {
              "id": "genres",
              "label": "Genre",
            },
          ]
        }
        selectedOptionId="title"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <ColorButton
        backgroundcolor="#ec407a"
        fontcolor="#fff"
        size="large"
        type="submit"
      >
        Search
      </ColorButton>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</form>
`;
